synapse/federation/federation_client.py

```diff
@@ -319,7 +323,13 @@ async def get_pdu_from_destination_raw(
             pdu = pdu_list[0]
 
             # Check signatures are correct.
-            signed_pdu = await self._check_sigs_and_hash(room_version, pdu)
+            try:
+                signed_pdu = await self._check_sigs_and_hash(room_version, pdu)
+            except InvalidEventSignatureError as e:
+                errmsg = f"event id {pdu.event_id}: {e}"
+                logger.warning("%s", errmsg)
+                raise SynapseError(403, errmsg, Codes.FORBIDDEN)
```

Comment:
this (and the three other copies of the same change) should be a non-functional change; we're just pulling the functionality out of `_check_sigs_and_hash`